packages/director/src/app.ts

```diff
@@ -95,6 +124,49 @@ app.put(
     console.log(`>> Received instance result`, { instanceId });
     await executionDriver.setInstanceResults(instanceId, result);
 
+    const instance = await executionDriver.getInstanceById(instanceId);
+    const run = await executionDriver.getRunWithSpecs(instance.runId);
+    const project = await executionDriver.getProjectById(run.meta.projectId);
+    const isRunStillRunning = run.specs.reduce(
+      (
+        wasRunning: boolean,
+        currentSpec: {
+          claimed: boolean;
+          results: any;
+        },
+        index: number
+      ) => {
+        return (
+          !currentSpec.claimed || !run.specsFull[index].results || wasRunning
+        );
+      },
+      false
+    );
```

Comment:
Should it be a distinct method?

Reply:
What alternative are you thinking of? 

I separated it into a function for readability. If thats what you are asking about.

Reply:
Yes, this